lib/rspec/support/source/node.rb

```diff
@@ -45,13 +45,18 @@ def location
           @location ||= args.find { |arg| arg.is_a?(Location) }
         end
 
-        def each(&block)
+        def each
```

Comment:
Given the lack of spec to cover this (which is completely understandable!), can you leave a comment on this method explaining why it uses a loop-based approach instead of recursion?